kernel/src/scan/mod.rs

```diff
@@ -215,26 +245,19 @@ impl Scan {
                     0
                 };
```

Comment:
aside: This code is more verbose than necessary because we don't unpack the `read_result`:
```rust
            for read_result in read_results {
                let len = if let Ok(ref res) = read_result {
                    res.length()
                } else {
                    0
                };
```
should probably be just:
```rust
            for read_result in read_results {
                let read_result = read_result?;
                let len = read_result.length();
```
However, looking deeper:
1. I think the code was originally trying to avoid converting an error `read_result` into an error result for the overall `execute` method
   * The computation of `len` carefully avoids `?` that would error-return the whole method
   * `ScanResult` takes a `DeltaResult<Box<dyn EngineData>>` instead of just `Box<dyn EngineData>`
1. But that original intention seems to have been undermined by
   * The method returns a `Vec` instead of an `Iterator` (so there's no benefit to tracking individual errors)
   * The partition column frobbing uses `read_result?`, which would short-circuit the whole method
   * The DV code (and `ScanResult` definition) is weird, because it takes a `Result` internally.

Overall, we should probably decide whether we want this code to be lazy, and change the code to be consistent with that objective. If we want lazy, then:
* Return `Iterator<DeltaResult<ScanResult>>`
* Don't use `?` inside the `read_result` loop
* `ScanResult::raw_result` is a `Box<dyn EngineData>`
* etc.

(but all this can be a separate PR, maybe we just add a tracking ticket?)

Reply:
Yeah, I think this is covered by #123 , but I'll link this comment there as more context